app/models/detailed_guide.rb

```diff
@@ -62,6 +65,35 @@ def related_detailed_guide_ids=(detailed_guide_ids)
     self.related_documents = other_related_documents + detailed_guide_documents
   end
 
+  def need_ids
+    return @need_ids if @need_ids
+    return [] unless document # TODO: Remove
```

Comment:
So I obviously missed the TODO, as that was meant for me... Anyway, looking again at this, the issue is that with new Editions, they do not have a document, and therefore, I don't think you can look up the corresponding needs.

I've moved this around a bit, and made the comment more useful. Updated patch here b8ae67778eda989243ee76cec6c59b41387f7e51 .